client/src/pages/book/BookDetails.tsx

```diff
@@ -64,14 +70,28 @@ function CategoryTable({ categories }: { categories: BookCategoryViewModelType[]
 }
 
 function BookDetails() {
-  const params = useParams();
-  const [bookId, setBookId] = useState<string>('');
+  const { register, handleSubmit } = useForm<ToggleBookInUserListType>({
+    resolver: zodResolver(ToggleBookInUserList),
+  });
+  const mutation = useMutation({
+    mutationFn: toggleBookInUserList,
+    onError: (e: Error) => {
+      console.log(e);
+    },
+    onSuccess: () => {
+      data!.doesUserObserve = !data!.doesUserObserve;
+    },
+  });
+  const onClick: SubmitHandler<ToggleBookInUserListType> = (toggleData) => {
+    mutation.mutate(toggleData);
+  };
 
+  const params = useParams();
   const { data, status } = useQuery({
     queryKey: ['books', params.bookId],
     queryFn: () => getBook(params.bookId + ''),
   });
-
+  const [bookId, setBookId] = useState<string>('');
   useEffect(() => {
     if (status === 'success' && data?.id) {
       setBookId(data.id);
```

Comment:
Tu też nie wiem który sposób lepszy, czy ten useEffect jest właściwie potrzebny, skoro bierzemy to z URLa?

Reply:
nie jest, powinniśmy bazować tylko na tym id co mamy w url

Reply:
Git, bo to było wcześniej dodane, to usuwam